python/composio/tools/toolset.py

```diff
@@ -1157,7 +1170,16 @@ def get_expected_params_for_user(
         """
         # If `integration_id` is provided, use it to fetch the params
         if integration_id is not None:
-            return self._get_expected_params_from_integration_id(id=integration_id)
+            response = self._get_expected_params_from_integration_id(id=integration_id)
+            if auth_scheme is not None and response["auth_scheme"] != auth_scheme:
+                raise ComposioSDKError(
+                    message=(
+                        "Auth scheme does not match provided integration ID, "
+                        f"auth scheme associated with integration ID {response['auth_scheme']} "
```

Comment:
The error message has been improved to include the auth scheme information. This will be helpful for debugging and providing clearer feedback to users. Good update!